src/Compilers/VisualBasic/Portable/Symbols/Source/SourceMethodSymbol.vb

```diff
@@ -1572,6 +1572,10 @@ lReportErrorOnTwoTokens:
                 arguments.Diagnostics.Add(ERRID.ERR_ExplicitTupleElementNamesAttribute, arguments.AttributeSyntaxOpt.Location)
             End If
 
+            If arguments.Attribute.IsTargetAttribute(Me, AttributeDescription.ModuleInitializerAttribute) Then
```

Comment:
>If arguments.Attribute.IsTargetAttribute(Me, AttributeDescription.ModuleInitializerAttribute) [](start = 12, length = 93)

It looks like this check should be moved into DecodeWellKnownAttributeAppliedToMethod. Please add a test with the attribute applied to a return.